packages/api/tests/test-kinesis-consumer.js

```diff
@@ -59,92 +57,125 @@ const disabledRuleParams = Object.assign({}, commonRuleParams, {
   state: 'DISABLED'
 });
 
+/**
+ * Callback used for testing
+ *
+ * @param {*} err - error
+ * @param {Object} object - object
+ * @returns {Object} object, if no error is thrown
+ */
 function testCallback(err, object) {
   if (err) throw err;
   return object;
-};
+}
+
+test.beforeEach(async (t) => {
+  t.context.templateBucket = randomString();
+  await s3().createBucket({ Bucket: t.context.templateBucket }).promise();
+
+  t.context.stateMachineArn = randomString();
 
-test.before(async () => {
-  sinon.stub(Rule, 'buildPayload').resolves(true);
-  await manager.createTable(tableName, ruleTableParams);
+  t.context.queueUrl = await createQueue();
+
+  t.context.messageTemplate = {
+    cumulus_meta: {
+      state_machine: t.context.stateMachineArn
+    },
+    meta: { queues: { startSF: t.context.queueUrl } }
+  };
+  const messageTemplateKey = `${randomString()}/template.json`;
+  await s3().putObject({
+    Bucket: t.context.templateBucket,
+    Key: messageTemplateKey,
+    Body: JSON.stringify(t.context.messageTemplate)
+  }).promise();
+
+  sinon.stub(Rule, 'buildPayload').callsFake((item) =>
+    Promise.resolve({
+      template: `s3://${t.context.templateBucket}/${messageTemplateKey}`,
+      provider: item.provider,
+      collection: item.collection,
+      meta: get(item, 'meta', {}),
+      payload: get(item, 'payload', {})
+    })
+  );
+
+  t.context.tableName = randomString();
```

Comment:
Is there a reason to use random strings and regenerate these for every test? Since these are constants it makes more sense to me to define them outside of the before block and I don't really see a reason for having them be randomly generated since we statically define them in our deployments.

Reply:
I was having a lot of issues with concurrency it seemed like and had the hardest time getting the queueing tests to work. This seemed to help.

Reply:
I see - ya the concurrency in ava can be frustrating. I was hoping use --serial would get around these types of issues entirely. :/